superset-frontend/src/explore/components/DataTableControl.jsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useMemo } from 'react';
+import { styled, t } from '@superset-ui/core';
+import { FormControl } from 'react-bootstrap';
+import Button from '../../components/Button';
+import RowCountLabel from './RowCountLabel';
+import {
+  applyFormattingToTabularData,
+  prepareCopyToClipboardTabularData,
+} from '../../utils/common';
+import CopyToClipboard from '../../components/CopyToClipboard';
+
+const CopyButton = styled(Button)`
+  padding: ${({ theme }) => theme.gridUnit / 2}px
+    ${({ theme }) => theme.gridUnit * 2.5}px;
+  font-size: ${({ theme }) => theme.typography.sizes.s}px;
+
+  // needed to override button's first-of-type margin: 0
+  && {
+    margin: 0 ${({ theme }) => theme.gridUnit * 2}px;
+  }
+
+  i {
+    padding: 0;
+  }
+`;
+
+export const CopyToClipboardButton = ({ data }) => (
+  <CopyToClipboard
+    text={data ? prepareCopyToClipboardTabularData(data) : ''}
+    wrapped={false}
+    copyNode={
+      <CopyButton>
+        <i className="fa fa-clipboard" />
+      </CopyButton>
+    }
+  />
+);
+
+export const FilterInput = ({ filterText, onChangeHandler }) => (
+  <FormControl
+    placeholder={t('Search')}
+    bsSize="sm"
+    value={filterText}
+    onChange={onChangeHandler}
+    style={{ paddingBottom: '5px' }}
```

Comment:
can we use `styled` to apply the css here (or at least get the constant padding from the superset theme)?

Reply:
I removed this padding, it wasn't needed